core/txpool/legacypool/legacypool.go

```diff
@@ -1693,13 +1880,19 @@ func (pool *LegacyPool) demoteUnexecutables() {
 			}
 			pendingGauge.Dec(int64(len(gapped)))
 		}
-		// Delete the entire pending entry if it became empty.
 		if list.Empty() {
+			// Delete the entire pending entry if it became empty.
 			delete(pool.pending, addr)
 			if _, ok := pool.queue[addr]; !ok {
 				pool.reserver.Release(addr)
 			}
+		} else {
+			// Update the latest known pending nonce
+			highestPending := list.LastElement()
+			nonces[addr] = highestPending.Nonce() + 1
+			pool.pendingNonces.setAll(nonces)
```

Comment:
This change was done earlier in outer function but I have moved it here to simple save extra iteration on the whole pending pool.